File: backend-fastapi/background/handlers/ai.py
import logging

from background.bgconfig import PlaceEmbbeddingData
from models.cruds import CrudsPlace
from services.instances import pg_client


async def place_embedding_task(payload: PlaceEmbbeddingData):
    async with pg_client.session() as session:
        cruds = CrudsPlace(session)
        place_id = payload["place_id"]
        result = await cruds.embed(place_id)
        logging.info(result)

File: backend-axum/src/background/handlers/ai.rs
use apalis::prelude::*;

use crate::{
    background::bgconfig::{AIJob, PlaceEmbeddingTask},
    lib_::clients::HandlerError,
    models::cruds::CrudsPlace,
    services::SharedState,
};

async fn handle_place_embedding(
    state: SharedState,
    payload: PlaceEmbeddingTask,
) -> Result<(), HandlerError> {
    let cruds = CrudsPlace::new(state);
    let result = cruds.embed(payload.id).await.map_err(|err| HandlerError {
        taskname: "place_embedding".to_string(),
        message: "womething went wrong when trying to update place embedding".to_string(),
        details: err.into_json(),
    })?;
    println!("{:?}", result);
    Ok(())
}

pub async fn handle_ai_tasks(job: AIJob, data: Data<SharedState>) -> Result<(), HandlerError> {
    let state = (*data).clone();
    match job {
        AIJob::PlaceEmbedding(payload) => handle_place_embedding(state, payload).await,
    }
}
